cdap-standalone/src/main/java/co/cask/cdap/TrackerAppCreationService.java

```diff
@@ -0,0 +1,179 @@
+import co.cask.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import co.cask.cdap.internal.app.services.ApplicationLifecycleService;
+import co.cask.cdap.internal.app.services.ProgramLifecycleService;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
+import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProgramId;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.AbstractExecutionThreadService;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Create Tracker App in Default Namespace and start its Programs during Standalone startup.
+ */
+public class TrackerAppCreationService extends AbstractExecutionThreadService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TrackerAppCreationService.class);
+  private static final Gson GSON = new Gson();
+  private final CConfiguration cConf;
+  private final ArtifactRepository artifactRepository;
+  private final ApplicationLifecycleService applicationLifecycleService;
+  private final ProgramLifecycleService programLifecycleService;
+  private final ApplicationId applicationId;
+  private final ProgramId auditFlow;
+  private final ProgramId auditService;
+  private ArtifactSummary artifactSummary;
+
+  @Inject
+  public TrackerAppCreationService(CConfiguration cConf, ArtifactRepository artifactRepository,
+                                   ApplicationLifecycleService applicationLifecycleService,
+                                   ProgramLifecycleService programLifecycleService) {
+    this.cConf = cConf;
+    this.artifactRepository = artifactRepository;
+    this.applicationLifecycleService = applicationLifecycleService;
+    this.programLifecycleService = programLifecycleService;
+    this.applicationId = new ApplicationId(NamespaceId.DEFAULT.getNamespace(), "_Tracker");
+    this.auditFlow = new ProgramId(applicationId.getNamespace(), applicationId.getApplication(), ProgramType.FLOW,
+                                   "AuditLogFlow");
+    this.auditService = new ProgramId(applicationId.getNamespace(), applicationId.getApplication(), ProgramType.SERVICE,
+                                      "AuditLog");
+  }
+
+  @Override
+  protected void run() throws Exception {
+    try {
+      // Wait until tracker artifact is available
+      Tasks.waitFor(true, new Callable<Boolean>() {
+        @Override
+        public Boolean call() throws Exception {
+          List<ArtifactSummary> artifacts = null;
+          try {
+            artifacts = artifactRepository.getArtifacts(NamespaceId.SYSTEM, "tracker");
+          } catch (ArtifactNotFoundException ex) {
```

Comment:
@chtyim Looks like it does throw this exception. 
   * @throws ArtifactNotFoundException if no artifacts of the given name in the given namespace exist